anomalydetection/detectors.py

```diff
@@ -0,0 +1,105 @@
+import pandas as pd
+
+
+class BaseDetector:
+    def __init__(self):
+        pass
+
+    def fit(self, data: pd.Series):
+        """ Set detector parameters based on data. """
+        self.validate(data)
+        return self
+
+    def detect(self, data: pd.Series):
+        """ Detect anomalies in data. """
+        return pd.Series([False] * len(data), index=data.index)
+
+    def validate(self, data):
+        if not isinstance(data, pd.Series):
+            raise ValueError("Input data must be a pandas.Series.")
+
+
+class AnomalyDetectionPipeline(BaseDetector):
+    """ Combine detectors. """
+    def __init__(self, detectors):
+        super().__init__()
+        self._detectors = detectors
+        self._series_name = "is_anomaly"
+
+    def fit(self, data):
+        for detector in self._detectors:
+            detector.fit(data)
+        return self
+
+    def detect(self, potentially_abnormal_data):
+        detected_anomalies = [False] * len(potentially_abnormal_data)
+        for detector in self._detectors:
+            detected_anomalies |= detector.detect(potentially_abnormal_data)
```

Comment:
Perhaps a voting strategy would work better? With more detectors, risk of false positive detections will increase

Reply:
So you are proposing &= instead ? Or changing from bool to int to count the votes?

Reply:
I moved to issue #3